tasks/scripting/src/main/java/org/infinispan/scripting/impl/DataTypedCacheManager.java

```diff
@@ -34,11 +38,13 @@ public DataTypedCacheManager(DataType dataType, Optional<Marshaller> marshaller,
 
    @Override
    public <K, V> Cache<K, V> getCache(String cacheName) {
```

Comment:
The encoder to use can be precalculated in the constructor:
```Class<? extends Encoder> encoderClass = dataType == DataType.UTF8 ? UTF8KeyValueEncoder.class :            marshaller.isPresent() ? GenericJbossMarshallerEncoder.class : IdentityEncoder.class;``` and then getCache will be reduced to a one-liner